sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -196,6 +196,84 @@
  *    <li>Since version: 3.4.0</li>
  *   </ul>
  *  </li>
+ *  <li>Name: <code>DATE_ADD</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>DATE_ADD(start_date, num_days)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>DATEDIFF</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>DATEDIFF(end_date, start_date)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>TRUNC</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>TRUNC(date, format)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>EXTRACT</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>EXTRACT(field, source, replacement)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>SECOND</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>SECOND(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>MINUTE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>MINUTE(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>HOUR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>HOUR(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>MONTH</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>MONTH(date)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>QUARTER</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>QUARTER(date)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>YEAR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>YEAR(date)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>DAYOFWEEK</code>
```

Comment:
The name seems not correct

Reply:
Refer http://www.h2database.com/html/functions.html#day_of_week
https://prestodb.io/docs/current/functions/datetime.html#day_of_week

Reply:
But the name is DAYOFWEEK in spark.
https://github.com/apache/spark/blob/2f8613f22c0750c00cf1dcfb2f31c431d8dc1be7/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L612

Reply:
I know.  `DAY_OF_WEEK` is style decision discussed offline between @cloud-fan and me.